sys/src/libc/port/cistrcmp.c

```diff
@@ -9,27 +9,27 @@
 
 #include <u.h>
 #include <libc.h>
+#include <ctype.h>
 
 int
 cistrcmp(const char *s1, const char *s2)
 {
-	int c1, c2;
+	unsigned int c1, c2;
 
-	while(*s1){
+	do{
 		c1 = *(uint8_t*)s1++;
 		c2 = *(uint8_t*)s2++;
-
 		if(c1 == c2)
 			continue;
+		if(isupper(c1))
```

Comment:
You are correct: I just checked toupper.c (which also has tolower.c in it) and it does a check itself so we should be good to go.